lib/source/mqtt/iot_mqtt_api.c

```diff
@@ -1117,6 +1099,8 @@ IotMqttError_t IotMqtt_Connect( const IotMqttNetworkInfo_t * pNetworkInfo,
         /* Set the MQTT packet serializer overrides. */
         #if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1
             pNewMqttConnection->pSerializer = pNetworkInfo->pMqttSerializer;
+        #else
+            pNewMqttConnection->pSerializer = NULL;
```

Comment:
Initializing the new `pSerializer` member when serializer overrides are _not_ enabled.